command/agent/config.go

```diff
@@ -711,7 +816,10 @@ func DecodeConfig(r io.Reader) (*Config, error) {
 	// use mapstructure decoding, so we need to account for those as well.
 	allowedKeys := []string{
 		"service", "services", "check", "checks", "statsd_addr", "statsite_addr", "statsite_prefix",
-		"dogstatsd_addr", "dogstatsd_tags",
+		"dogstatsd_addr", "dogstatsd_tags", "circonus_api_token", "circonus_api_app",
+		"circonus_api_url", "circonus_submission_url", "circonus_submit_interval",
```

Comment:
Ah ok this code was created to support having the keys at the top level in the old format, but the new structure puts all the metrics-related keys inside the `telemetry` sub-structure. Since Circonus is new we don't have any old format to support. You can rip out the `allowedKeys` changes as well as all the `if sub, ok...` stuff and put your config_test JSON inside a `telemetry` object and all should be good.